lightwood/encoders/categorical/autoencoder.py

```diff
@@ -43,6 +43,17 @@ def _encoder_targets(self, data):
         labels = targets_c.to(self.net.device)
         return labels
 
+    def to(self, device, available_devices):
+        if self.use_autoencoder:
+            self.net.to(device, available_devices)
+
+            modules = [module for module in self.net.modules() if type(
+                module) != torch.nn.Sequential and type(module) != DefaultNet]
+
+            self.encoder = torch.nn.Sequential(*modules[0:2])
```

Comment:
does this really belong in the to method? it seems like this should not be part of the method that defines what device to send it to, as its defining also the architecture of the encoder

Reply:
Well, it's not defining the architecture per say.

As in `self.net` is the underlying network and then `self.encoder` and `self.decoder` are just Sequential wrappers over part of that.

But honestly, I think there *might* be a better way of doing this, in that I might be able to get rid of self.encoder and self.decoder all together, I'll take a look at doing this tomorrow.

Reply:
lets keep it very standard, network element definition in the init, network/graph stitching in the forward.

Reply:
The reason why I was re-defining `self.encoder` and `self.decoder` is that I thought they might essentially become pointers to some null memory or cause a copy of the layers they are pointing to once we applied the `.to` method to the network above, but apparently I miss-understood how `torch.nn.Sequential` works and this is not the case, they keep pointing to the newly GPU/CPU-transfered layers.

Removed all the extra stuff and now this method is much simpler :)